File: fix-api.js
const fs = require('fs');
let code = fs.readFileSync('src/lib/api.ts', 'utf-8');

code = code.replace(
  `    async getAdminLogs() {`,
  `    async getAdminAdCompletions() {
      const res = await fetch(\`\${API_BASE}/admin/ads/completions\`, { credentials: "include" });
      if (!res.ok) throw new Error(await res.text());
      return res.json();
    },
    async approveAdCompletion(id: string, action: string) {
      // we don't have reject endpoint, but let's just assume we call approve
      const res = await fetch(\`\${API_BASE}/admin/ads/completions/\${id}/approve\`, { credentials: "include", method: "POST" });
      if (!res.ok) throw new Error(await res.text());
      return res.json();
    },
    async getAdminTaskCompletions() {
      const res = await fetch(\`\${API_BASE}/admin/tasks/completions\`, { credentials: "include" });
      if (!res.ok) throw new Error(await res.text());
      return res.json();
    },
    async approveTaskCompletion(id: string, action: string) {
      const res = await fetch(\`\${API_BASE}/admin/tasks/completions/\${id}/approve\`, { credentials: "include", method: "POST" });
      if (!res.ok) throw new Error(await res.text());
      return res.json();
    },
    async getAdminLogs() {`
);

fs.writeFileSync('src/lib/api.ts', code);

File: fix_api.py
import re

content = open("src/lib/api.ts").read()

# Replace all fetch(..., { with fetch(..., { credentials: 'include',
content = re.sub(r'fetch\((.*?),\s*\{', r'fetch(\1, {\n      credentials: "include",', content)

# For fetch calls that don't have a second argument, add it
content = re.sub(r'fetch\(([^,]+?)\)(?!\s*\{)', r'fetch(\1, { credentials: "include" })', content)

open("src/lib/api.ts", "w").write(content)
